docs/puzzles/pooling.md

```diff
@@ -0,0 +1,320 @@
+
+`Truths` is the data structure that is forcefully added to this puzzle as part of the singleton top layer.  It contains some data we will use in the puzzle which we will access by using some functions from the `singleton_truths.clib` library.
+
+There are two spend cases for this puzzle.  The **absorb case** will most likely be triggered by the pool and is how they claim the rewards that the farmers receive.  The **escape case** will be initiated by the farmer to begin the process of leaving the pool by heading to the waiting room.  The following two arguments change depending on the case we are executing:
+
+`p1` is named the way it is because it is going to be different depending on the spend type we are using.  It is either:
+ * The amount of the pool reward that is being claimed (1750000000000 in the first three years)
+ * A list of key value pairs that is used to reveal important information to the blockchain for wallets to use (similar to the [singleton launcher](https://chialisp.com/docs/puzzles/singletons#the-launcher))
+
+`pool_reward_height` is also different depending on the spend case, but in the escape case it is just `()` so we leave it named the way it is.  In the absorb case, it is the height of the pool reward that is being absorbed.  This is used along with `POOL_REWARD_PREFIX` to calculate the `parent_coin_info` of the reward coin so that we can calculate its ID.
+
+Let's talk about our main entry point:
+
+```chialisp
+(if pool_reward_height
+  (absorb_pool_reward POOL_PUZZLE_HASH
+                      (my_inner_puzzle_hash_truth Truths)
+                      (my_amount_truth Truths)
+                      p1
+                      (calculate_pool_reward pool_reward_height P2_SINGLETON_PUZZLE_HASH POOL_REWARD_PREFIX p1)
+  )
+  (travel_to_waitingroom OWNER_PUBKEY WAITINGROOM_PUZHASH (my_amount_truth Truths) p1)
+)
+```
+
+The main control flow here is whether or not the height is passed in as `()`. This is how we determine the spend type to execute. After that we just head to the appropriate function that generates the conditions we will need for the type of spend we are executing. You'll notice that we are extracting the relevant data from the `Truths` object before we pass them to the inner functions.
+
+We are also calculating one additional piece of information.  Let's look at it now:
+
+```chialisp
+(defun-inline calculate_pool_reward (pool_reward_height P2_SINGLETON_PUZZLE_HASH POOL_REWARD_PREFIX pool_reward_amount)
+  (sha256 (logior POOL_REWARD_PREFIX (logand (- (lsh (q . 1) (q . 128)) (q . 1)) pool_reward_height)) P2_SINGLETON_PUZZLE_HASH pool_reward_amount)
+)
+```
+
+This line may look complex, but it's doing something fairly simplistic.  It is calculating the coin ID of the reward coin it is claiming by manually calculating the parent info from the `POOL_REWARD_PREFIX` and `pool_reward_height`.
```

Comment:
This "parent info" isn't referring to parent coin is, right? (Since that's being renamed right now)